pyne/vta.py

```diff
@@ -0,0 +1,481 @@
+make sure all the wanted voxels will be found. Combining all the found voxels
+for each ray, we get the voxels intersecting with the facet.
+
+These voxels will be used to construct the bounding volume for source sampling.
+But currently these functions are separate from MOAB. Interface of MOAB with
+PyNE is still in need.
+
+.. moduleauthor:: X. Zhang
+"""
+from __future__ import division
+import math
+import numpy as np
+
+
+##################################
+### Facet voxel traverse tools ###
+##################################
+
+
+def _distance(start, end):
+    """
+    Calculate the distance between two points.
+
+    Parameters:
+    -----------
+    start : numpy array 
+        An array of size 3, represents the start point of  the line segment
+    end : numpy array
+        An array of size 3, represents the end point of  the line segment
+
+    Return:
+    -------
+    dist : float
+        The distance between the start point and the end point.
+    """
+    dist = math.sqrt((end[0] - start[0]) * (end[0] - start[0]) +
+                     (end[1] - start[1]) * (end[1] - start[1]) +
+                     (end[2] - start[2]) * (end[2] - start[2]))
+    return dist
+
+def _is_point_in_mesh(bounds, point):
+    """
+    Check whether a point is in the mesh.
+
+    Parameters:
+    -----------
+    bounds : list
+        Boundaries. Nested list of the boundaries along x, y, and z directions.
+        Expected format: [[x_min, ..., x_max],
+                          [y_min, ..., y_max],
+                          [z_min, ..., z_max]]
+    point : numpy array
+        An array with size of 3, represents the point to be checked.
+
+    Return:
+    -------
+    True : when point in the mesh.
+    False : when point outside the mesh.
+    """
+    if bounds[0][0] <= point[0] <= bounds[0][-1] and \
+       bounds[1][0] <= point[1] <= bounds[1][-1] and \
+       bounds[2][0] <= point[2] <= bounds[2][-1]:
+        return True
+    else:
+        return False
+
+def _find_voxel_idx_1d(bound, cor, v_1d):
+    """
+    Find the voxel id (in a given dimension).
+
+    Parameters:
+    -----------
+    bound : list
+        Boundary of the mesh in a given direction.
+    cor : float
+        Coordinate of the point in the given direction.
+    v_1d : float
+        Vector attribute in the given direction.
+
+    Return:
+    -------
+    idx : int
+        Mesh index of the given direction. This value is -1 when the point not
+        in the mesh.
+    ----------
+    """
+    idx = -1
+    if v_1d > 0:
+        for i in range(len(bound)-1):
+            if bound[i] <= cor < bound[i+1]:
+                idx = i
+                break
+    else:
+        for i in range(len(bound)-1, -1, -1):
+            if bound[i] < cor <= bound[i+1]:
+                idx = i
+                break
+    # idx could be -1, that means point outside the mesh
+    return idx
+
+def _cal_dir_v(start, end):
```

Comment:
Could we afford to be a little more verbose here? Even `_cal_dir_v` --> `_calc_vec_dir` provides a more clarity.